functions/controllers/cruzpoints/index.js

```diff
@@ -0,0 +1,81 @@
+const functions = require("firebase-functions");
+const express = require("express");
+const cors = require("cors");
+const { jwtCheck, hasCreateAdmin, hasUpdateHacker } = require("../../utils/middleware");
+const { setDocument, queryDocument, transaction } = require("../../utils/database");
+const { customAlphabet } = require("nanoid");
+
+const nanoid = customAlphabet("ABCDEFGHIJKLMNOPQRSTUVWXYZ123456789");
+
+const cruzpoints = express();
+cruzpoints.disable("x-powered-by");
+cruzpoints.use(express.json());
+
+const auth0Config = functions.config().auth;
+const corsConfig = auth0Config ? auth0Config.cors : "";
+
+const corsOptions = {
+  origin: corsConfig,
+  optionsSuccessStatus: 200,
+};
+
+cruzpoints.use(cors(corsOptions));
+
+cruzpoints.post("/createActivity", jwtCheck, hasCreateAdmin, async (req, res) => {
+  try {
+    const code = nanoid(7);
+    if (typeof req.body.points !== "number") {
+      res.status(500).send("Invalid Body Type");
```

Comment:
```suggestion
      res.status(500).send({ status: 500, message: "Invalid Body Type" });
```

Reply:
Generally for errors, the format is {status: StatusCode, error: ErrorMessage}, will make the status code change tho